qa/standalone/crush/erasure-code-profile.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2020 Red Hat <contact@redhat.com>
+#
+# Author: Prashant Dhange <pdhange@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Library Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library Public License for more details.
+#
+
+source $CEPH_ROOT/qa/standalone/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:7132" # git grep '\<7132\>' : there must be only one
+    export CEPH_ARGS
+    CEPH_ARGS+="--fsid=$(uuidgen) --auth-supported=none "
+    CEPH_ARGS+="--mon-host=$CEPH_MON "
+
+    local funcs=${@:-$(set | sed -n -e 's/^\(TEST_[0-9a-z_]*\) .*/\1/p')}
+    for func in $funcs ; do
+        setup $dir || return 1
+        $func $dir || return 1
+        teardown $dir || return 1
+    done
+}
+
+function TEST_valid_failuredomain() {
+    local dir=$1
+
+    run_mon $dir a || return 1
+
+    local profile=validecprofile
+
+    crush_domain=datacenter
+    ceph osd erasure-code-profile set $profile k=4 m=2 crush-failure-domain=$crush_domain 2>&1 || return 1
```

Comment:
i don't think this test belongs to the directory of `qa/standalone/crush`. as it is exercising `osd erasure-code-profile` handling. actually quite a few tests in this file have already been covered by `qa/standalone/mon/osd-erasure-code-profile.sh`. i see no reason to repeat them.

Reply:
In `qa/standalone/mon/osd-erasure-code-profile.sh`, we are setting crush failure domain to osd and not testing for other failure domains. Also whether profile reflecting correct failure domain. I am happy to remove this test from here.